velox/dwio/parquet/reader/ParquetData.cpp

```diff
@@ -17,15 +17,62 @@
 #include "velox/dwio/parquet/reader/ParquetData.h"
 
 #include "velox/dwio/common/BufferedInput.h"
+#include "velox/dwio/parquet/common/ParquetBloomFilter.h"
 #include "velox/dwio/parquet/reader/ParquetStatsContext.h"
 
 namespace facebook::velox::parquet {
 
+using thrift::RowGroup;
+
+namespace {
+bool isFilterRangeCoversStatsRange(
+    common::Filter* filter,
+    dwio::common::ColumnStatistics* stats,
+    const TypePtr& type) {
+  switch (type->kind()) {
+    case TypeKind::BIGINT:
+    case TypeKind::INTEGER:
+    case TypeKind::SMALLINT:
+    case TypeKind::TINYINT: {
+      auto intStats =
+          dynamic_cast<dwio::common::IntegerColumnStatistics*>(stats);
+      if (!intStats)
+        return false;
+
+      int64_t min =
+          intStats->getMinimum().value_or(std::numeric_limits<int64_t>::min());
+      int64_t max =
+          intStats->getMaximum().value_or(std::numeric_limits<int64_t>::max());
+
+      switch (filter->kind()) {
+        case common::FilterKind::kBigintRange:
+          return static_cast<common::BigintRange*>(filter)->lower() <= min &&
+              max <= static_cast<common::BigintRange*>(filter)->upper();
+        case common::FilterKind::kBigintMultiRange: {
+          common::BigintMultiRange* multiRangeFilter =
+              static_cast<common::BigintMultiRange*>(filter);
+          auto numRanges = multiRangeFilter->ranges().size();
+          if (numRanges > 0) {
+            return multiRangeFilter->ranges()[0]->lower() <= min &&
+                max <= multiRangeFilter->ranges()[numRanges - 1]->upper();
```

Comment:
Are the ranges must be sorted? If not, the `multiRangeFilter->ranges()[0]->lower()` will not be the least value.

Reply:
The check in the BigintMultiRange constructor suggests that the ranges are sorted.